src/facet_display/FacetDisplay.tsx

```diff
@@ -0,0 +1,132 @@
+import React from "react"
+import FilterableFacet from "./FilterableFacet"
+import Facet from "./Facet"
+import SearchFilter from "./SearchFilter"
+import type { FacetManifest, Facets, Aggregation, Bucket } from "./types"
+import { LEVELS, DEPARTMENTS } from "../constants"
+
+export type BucketWithLabel = Bucket & { label: string }
+
+interface Props {
+  facetMap: FacetManifest
+  facetOptions: (group: string) => Aggregation | null
+  activeFacets: Facets
+  clearAllFilters: () => void
+  onFacetChange: (name: string, value: string, isEnabled: boolean) => void
+}
+
+export const getDepartmentName = (departmentId: string): string => {
+  if (departmentId in DEPARTMENTS) {
+    return DEPARTMENTS[departmentId as keyof typeof DEPARTMENTS]
+  } else {
+    return departmentId
+  }
+}
+
+export const getLevelName = (levelValue: string): string => {
+  if (levelValue in LEVELS) {
+    return LEVELS[levelValue as keyof typeof LEVELS]
+  } else {
+    return levelValue
+  }
+}
+
+const resultsWithLabels = (
+  results: Aggregation | null,
+  labelFunction: ((value: string) => string) | null | undefined
+): BucketWithLabel[] => {
+  const newResults = [] as BucketWithLabel[]
+  ;(results || []).map((singleFacet: Bucket) => {
```

Comment:
Let's default the argument here to avoid the leading semi, ie. `results: Aggregation = [],`, though we may need non-null assertions below, `!`.